src/main/java/org/jabref/logic/bibtex/DuplicateCheck.java

```diff
@@ -303,6 +305,15 @@ public boolean isDuplicate(final BibEntry one, final BibEntry two, final BibData
             return true;
         }
 
+        // check DOI
+        if (one.getDOI().isPresent() && two.getDOI().isPresent() && one.getDOI().get().isCompareNotExact(two.getDOI().get())) {
```

Comment:
I would suggest calling the Doi parse method first on both sides and then compare them.
The Doi parse method can already parse different doi formats e.g. As url or in the short form.

For ISBN the same. I think the isbn can parse both isbn 10 and isbn 13

Reply:
Isn't this what `getDoi` is doing?
https://github.com/JabRef/jabref/blob/7ddfd9c49a4b64edefe60f3027a044fc5a640a25/src/main/java/org/jabref/model/entry/BibEntry.java#L458-L460

Reply:
Ah my fault, oversaw that the parse is already called.

Reply:
I rewrote it. I reduced the amount of parse().